src/index.js

```diff
@@ -4,189 +4,150 @@
 
 'use strict'
 
-const fs = require('fs')
-const multihashing = require('multihashing')
-const base58 = require('bs58')
-const forge = require('node-forge')
-const protobuf = require('protocol-buffers')
-const path = require('path')
-
-const pbCrypto = protobuf(fs.readFileSync(path.resolve(__dirname, '../protos/crypto.proto')))
+const mh = require('multihashes')
+const crypto = require('libp2p-crypto')
+const assert = require('assert')
+
+class PeerId {
+  constructor (id, privKey, pubKey) {
+    if (Buffer.isBuffer(id)) {
+      this.id = id
+    } else {
+      throw new Error('invalid id provided')
+    }
 
-exports = module.exports = PeerId
+    if (pubKey) {
+      assert(this.id.equals(pubKey.hash()), 'inconsistent arguments')
+    }
 
-exports.Buffer = Buffer
+    if (privKey) {
+      assert(this.id.equals(privKey.public.hash()), 'inconsistent arguments')
+    }
 
-function PeerId (id, privKey, pubKey) {
-  const self = this
+    if (privKey && pubKey) {
+      assert(privKey.public.bytes.equals(pubKey.bytes), 'inconsistent arguments')
+    }
 
-  if (!(self instanceof PeerId)) {
-    throw new Error('Id must be called with new')
+    this.privKey = privKey
+    this._pubKey = pubKey
   }
 
-  self.privKey = privKey
-  self.pubKey = pubKey
-  self.id = id // multihash - sha256 - buffer
+  get pubKey () {
+    if (this._pubKey) {
+      return this._pubKey
+    }
 
-  // pretty print
-  self.toPrint = function () {
-    return {
-      id: self.toB58String(),
-      privKey: privKey.toString('hex'),
-      pubKey: pubKey.toString('hex')
+    if (this.privKey) {
+      return this.privKey.public
```

Comment:
This is strange. Why would we store the public key inside the privKey? In which case don't we update the pubKey prop when adding a privKey?